Undeprecate projects

danburonline opened the pull request:
Fixes [#3908](https://github.com/BlueBrain/nexus/issues/3908)

## Description

![Screenshot](https://github.com/BlueBrain/nexus-web/assets/34251194/a122c694-b788-4628-ac0b-d4a21d09eaec)

- Add an "Undo Deprecation" button.
-  Include icons for the other buttons.
-  Refactor and clean up code.
-  The behaviour has been modified for "deprecation" (destructive) and "undoing deprecation" (non-destructive). Currently, when you deprecate a project, you will be redirected to the project overview page (still the same). On the other hand, when you undo a deprecation, the state changes, and you remain on the settings page/view. This ensures a smoother flow compared to going back to the project overview.

## How has this been tested?

- Production and development builds locally on my MacBook.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have added screenshots (if applicable), in the comment section.

bilalesi commented:
My first impression is the un-depreacate button, can we use the same as deprecation button, one button can do both @danburonline, I will left some comments regarding code

bilalesi commented on src/subapps/admin/components/Settings/DangerZoneSubView.tsx:
```diff
@@ -130,6 +161,39 @@ const DangerZoneSubView = ({ project }: Props) => {
     }
   );
+
+  const undoDeprecateProjectAsync = async (
+    apiEndpoint: string,
+    nexus: NexusClient,
```
I suggest to use `useMutation` from react-query 
1. get the loading state pass it to the action button
2. can handle success/failure part more clean
3. if we are fetching the project details with RQ, then you can do the refresh just by invalidating the cache

danburonline replied:
Done.

bilalesi commented on src/subapps/admin/components/Settings/DangerZoneSubView.tsx:
```diff
@@ -233,11 +303,50 @@ const DangerZoneSubView = ({ project }: Props) => {
+            type="primary"
+            htmlType="button"
+            disabled={!project._deprecated} // Enable button only if project is deprecated
+            onClick={() =>
```
the onClick can be pure fn since it's has unique use case and one place, meaning, it can take only fn without input data since the data is reside inside the component

danburonline commented:
> My first impression is the un-depreacate button, can we use the same as deprecation button, one button can do both @danburonline, I will left some comments regarding code

Great feedback! I have implemented the changes and updated the data logic to utilise React Query for the project settings. Take a look.

bilalesi commented on src/pages/ProjectPage/ProjectPage.tsx:
```diff
@@ -42,26 +41,39 @@ const ProjectView: React.FunctionComponent = () => {
+  const { data: project, error } = useQuery<ProjectResponseCommon, Error>(
+    ['project', orgLabel, projectLabel],
+    () => fetchProjectData(orgLabel, projectLabel),
+    {}
```
you can remove the options object, if you do not need it

bilalesi commented on src/pages/ProjectPage/ProjectPage.tsx:
```diff
@@ -42,26 +41,39 @@ const ProjectView: React.FunctionComponent = () => {
   } = match;
 
+  const { data: project, error } = useQuery<ProjectResponseCommon, Error>(
+    ['project', orgLabel, projectLabel],
```
👍

bilalesi commented on src/subapps/admin/components/Settings/DangerZoneSubView.tsx:
```diff
@@ -130,6 +158,42 @@ const DangerZoneSubView = ({ project }: Props) => {
+    {
+      onSuccess: () => {
+        // Invalidate project query to refetch the project to update the UI
+        queryClient.invalidateQueries(['project', orgLabel, projectLabel]);
```
👍

bilalesi commented on src/subapps/admin/components/Settings/DangerZoneSubView.tsx:
```diff
@@ -246,7 +345,7 @@ const DangerZoneSubView = ({ project }: Props) => {
         <div className="settings-view-container">
           <div className="danger-actions">
-            <List<TDangerZoneItem>
+            <List<DangerZoneItem>
```
can we keep previous types and names of any vars as it is for the moment until we prepare some naming/typing convention, it's not right to rename as everyone prefer, 
as long as the naming do not has typos, will keep it

danburonline replied:
Reverted it.

bilalesi reviewed (approved):
I left a minor comment (feel free to discard it for the moment), and another regarding the naming convention 

nice work 👍
